packages/embark-code-runner/src/index.ts

```diff
@@ -52,101 +50,57 @@ export default class CodeRunner {
-
-  this.events.on("runcode:init-console-code:updated", (code: string, cb: Callback<null>) => {
-    this.evalCode(code, (err, _result) => {
-      if (err) {
-        this.logger.error("Error running init console code: ", err.message || err);
-      }
-      this.fireEmbarkJSEvents(code);
-      cb();
-    });
-  });
+    this.events.on("runcode:register", this.registerVar.bind(this));
+  }
 
-  this.events.on("runcode:embarkjs-code:updated", (code: string, cb: Callback<any>) => {
-    this.evalCode(code, (err, _result) => {
-      if (err) {
-        this.logger.error("Error running embarkjs code: ", err.message || err);
-      }
-      this.fireEmbarkJSEvents(code);
-      cb();
+  private registerCommands() {
+    this.events.setCommandHandler("runcode:getContext", (cb) => {
+      cb(this.vm.options.sandbox);
     });
-  });
-}
+    this.events.setCommandHandler("runcode:eval", this.evalCode.bind(this));
+    this.events.setCommandHandler("runcode:embarkjs:reset", this.resetEmbarkJS.bind(this));
+    this.events.on("contractsDeployed", this.resetEmbarkJS.bind(this));
+
+    // register listeners for when EmbarkJS runs registerProvider through the console.
+    // For example, when `EmbarkJS.Storage.registerProvider(...)` is run through the console,
+    // emit the `runcode:storage:providerRegistered` event, and fire any requests attached to it
+    Object.keys(EmbarkJS)
+      .filter((propName) => EmbarkJS[propName].hasOwnProperty("registerProvider"))
+      .forEach((providerName) => {
+        this.generateListener(providerName.toLowerCase(), ProviderEventType.ProviderRegistered);
+      });
 
-private registerCommands() {
-  this.events.setCommandHandler("runcode:getContext", (cb) => {
-    cb(this.vm.options.sandbox);
-  });
-  this.events.setCommandHandler("runcode:eval", this.evalCode.bind(this));
-  this.events.setCommandHandler("runcode:ready", (cb) => {
-    if (this.ready) {
-      return cb();
-    }
-    this.events.once("runcode:ready", cb);
-  });
-  this.events.setCommandHandler("runcode:embarkjs:reset", this.resetEmbarkJS.bind(this));
-
-  // register listeners for when EmbarkJS runs registerProvider through the console.
-  // For example, when `EmbarkJS.Storage.registerProvider(...)` is run through the console,
-  // emit the `runcode:storage:providerRegistered` event, and fire any requests attached to it
-  Object.keys(EmbarkJS)
-    .filter((propName) => EmbarkJS[propName].hasOwnProperty("registerProvider"))
-    .forEach((providerName) => {
-      this.generateListener(providerName.toLowerCase(), ProviderEventType.ProviderRegistered);
-    });
+    // register listeners for when EmbarkJS runs setProvider through the console.
+    // For example, when `EmbarkJS.Storage.setProvider(...)` is run through the console,
+    // emit the `runcode:storage:providerSet` event, and fire any requests attached to it
+    Object.keys(EmbarkJS)
+      .filter((propName) => EmbarkJS[propName].hasOwnProperty("setProvider"))
```

Comment:
Very similar to the comment above, `hasOwnProperty()` is unnecessary here.

Reply:
See: https://github.com/embark-framework/embark/pull/1680#discussion_r297222775.